src/main/java/at/hannibal2/skyhanni/features/garden/inventory/GardenPlotIcon.kt

```diff
@@ -0,0 +1,136 @@
+import at.hannibal2.skyhanni.events.InventoryOpenEvent
+import at.hannibal2.skyhanni.features.garden.GardenAPI
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils.chat
+import io.github.moulberry.notenoughupdates.events.ReplaceItemEvent
+import io.github.moulberry.notenoughupdates.events.SlotClickEvent
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.client.player.inventory.ContainerLocalMenu
+import net.minecraft.init.Blocks
+import net.minecraft.init.Items
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class GardenPlotIcon {
+
+    private val config get() = SkyHanniMod.feature.garden.plotIcon
+    private var showItem = false
+    private var copyStack: ItemStack? = null
+    private var editMode = 0 // 0 = off, 1 = on, 2 = reset
+    private var lastClickedSlotId = -1
+    private var originalStack = mutableMapOf<Int, ItemStack>()
+
+    companion object {
+        var hardReset = false
+        private val plotList get() = GardenAPI.config?.plotIcon?.plotList
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        for ((index, stack) in event.inventoryItems){
+            originalStack[index] = stack
+        }
+        showItem = GardenAPI.inGarden() && event.inventoryName == "Configure Plots"
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        showItem = false
+        editMode = 0
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun replaceItem(event: ReplaceItemEvent) {
+        if (!config.enabled) return
+        if (!showItem) return
+        val plotList = GardenPlotIcon.plotList ?: return
+
+        if (hardReset) {
+            plotList.clear()
+            hardReset = false
+            return
+        }
+
+        if (event.inventory is ContainerLocalMenu) {
+            if (event.slotNumber == 53) {
+                val mode: String = when (editMode) {
+                    0 -> "§cOFF"
+                    1 -> "§aON"
+                    2 -> "§9RESET"
+                    else -> ""
+                }
+                event.replaceWith(Utils.createItemStack(ItemStack(Items.wooden_axe).item, "§6Edit Mode [$mode§6]", "", "§bSwitch Edit Mode"))
+            }
+
+            if (plotList.isNotEmpty() && plotList.contains(event.slotNumber)) {
+                val old = originalStack[event.slotNumber]?: return
+                val new = event.slotNumber.getItem()?: return
+                val name = old.displayName
+                val lore = old.getLore().toTypedArray()
+                if (lastClickedSlotId == event.slotNumber) {
+                    lastClickedSlotId = -1
+                    return
+                }
+                val replaceStack = Utils.editItemStackInfo(new, name, true, *lore)
```

Comment:
There, i have no idea why patcher override the new name with the name of the stack you clicked in your inventory